### PR TITLE
[V2.0] Docs update

### PR DESCRIPTION
**Pull request by DyrellC**
# Description of change
Updating documentation across all crates. 
- [ ] wiki 
- [ ] readme's 
- [x] streams
- [x] lets
- [x] spongos

## Type of change
- Documentation Fix

### CODE REVIEW

File: streams/src/api/user_builder.rs [hunk 1]
```diff
@@ -117,14 +117,9 @@ impl<T> UserBuilder<T> {
...
     ///
-    /// If a [`Transport`] is not provided the builder will use a default client
-    /// ([`Client`](streams_app::transport::tangle::client::Client) at <https://chrysalis-nodes.iota.org>
-    /// if the `tangle` feature is enabled,
-    /// [`BucketTransport`](streams_app::transport::BucketTransport) if not)
-    ///
-    /// # Errors
-    /// This function will error out if the [`UserIdentity`] parameter is missing, as this makes up
-    /// the essence of a [`User`] and is required for any use case.
+    /// If a [`Transport`] is not provided the builder will use a default client.
```
Comment (kwek20) [inline streams/src/api/user_builder.rs:120]:
```suggestion
    /// If a [`Transport`] is not provided, the builder will use a default client.
```

File: streams/src/message/unsubscription.rs [hunk 2]
```diff
@@ -84,23 +82,32 @@ where
...
+    /// # Arguments:
+    /// * `initial_state`: The initial [`Spongos`] state the message will be joined to
     pub(crate) fn new(initial_state: &'a mut Spongos) -> Self {
         Self {
             initial_state,
             subscriber_id: Identifier::default(),
         }
     }
 
+    /// Returns a reference to the [`Identifier`] of the subsriber
```
Comment (kwek20) [inline streams/src/message/unsubscription.rs:105]:
```suggestion
    /// Returns a reference to the [`Identifier`] of the subscriber
```

**Review by kwek20 — CHANGES_REQUESTED** [review]

File: lets/src/message/version.rs [hunk 3]
```diff
@@ -1,23 +1,27 @@
...
 //! versioning (ie. include `content_version` field for example).
 //!
-//! Streams message syntax version is indicated as the first tryte in the trinary encoded message.
+//! Streams message syntax version is indicated as the first byte in the binary encoded message.
 //!
 //! Backwards compatibility of the Streams implementations is welcome and not mandatory.
 
 /// Streams version number.
-pub(crate) const STREAMS_1_VER: u8 = 0;
+pub(crate) const STREAMS_1_VER: u8 = 1;
```
[the remark this answers is not in the document]
Comment (DyrellC) [reply lets/src/message/version.rs:15]: I was running on habit to not update versions until we're pushing the release

File: streams/src/api/mod.rs [hunk 4]
```diff
@@ -2,16 +2,16 @@
 mod cursor_store;
 
 /// Unwrapped Message Types
-pub(crate) mod message;
+pub mod message;
 /// Message builder for sending payloads
-mod message_builder;
+pub mod message_builder;
```
[the remark this answers is not in the document]
Comment (DyrellC) [reply streams/src/api/mod.rs:7]: It is, and I agree.

[the remark this answers is not in the document]
Comment (DyrellC) [reply streams/src/api/mod.rs:7]: Alternative is to remove all links to `MessageBuilder` and `Messages` which makes the `User` docs more rigid

File: lets/src/message/version.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (kwek20) [reply lets/src/message/version.rs:15]: Fair enough!

**Review by kwek20 — APPROVED** [review]
LGTM! Whats the plan with wiki and readme? we will do that in another pr?

**Pull request merged**